core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -1864,13 +1873,82 @@ private void openReaderAndEnumerateColumns(SqlExecutionContext executionContext,
             model.setTableVersion(r.getVersion());
             model.setTableId(r.getMetadata().getId());
             copyColumnsFromMetadata(model, r.getMetadata(), false);
+            if (model.isUpdate()) {
```

Comment:
can we possibly reuse fields copied just above?

Reply:
When update goes on we want to copy all column names and types, not only queried. Column names and types copied from metadata. I can optimise to not copy all the columns but only those which will be written to leave a todo to do it later since I have a feeling it won't be most expensive part of the query

Reply:
We spoke about this and since we read all the columns anyway the optimisation I mentioned does not make sense to do for updates only. Updates also need types data which is not necessary for selects atm and we agreed to move types into select query model columns is significant refactoring with no obvious benefit

Reply:
agreed